### PR TITLE
Remove unnecessary null check

### PR DESCRIPTION
**Pull request by KaoruToda**
con is always NULL.

### CODE REVIEW

File: apps/s_server.c [hunk 1]
```diff
@@ -2202,22 +2202,21 @@ static int sv_body(int s, int stype, int prot, unsigned char *context)
             BIO_printf(bio_err, "Turned on non blocking io\n");
     }
 
-    if (con == NULL) {
-        con = SSL_new(ctx);
+    con = SSL_new(ctx);
```
Comment (richsalz) [inline apps/s_server.c]: Please add a check for NULL return; probably just as simple as
```
if (con == NULL)
    goto err;
```

Comment (KaoruToda) [reply apps/s_server.c]: @richsalz Thank you for your review and comment.
I add error check for SSL_new, then I sent new pull request(#4560).

Comment (KaoruToda) [reply apps/s_server.c]: @richsalz I canceled #4560, because I perceived that the error handling was not sufficient.
I'm going to send new request later.

Comment (kroeckx) [reply apps/s_server.c]: You know you can just add (or replace) patches in an existing pull request? Just add them to your branch and push that branch to github.

Comment (KaoruToda) [reply apps/s_server.c]: @kroeckx Thank you for your information. Add new code my same branch.

@richsalz I add a check for NULL return. I found same problem in s_client.c and fixed it.

File: apps/s_server.c [hunk 2]
```diff
@@ -2202,22 +2202,25 @@ static int sv_body(int s, int stype, int prot, unsigned char *context)
             BIO_printf(bio_err, "Turned on non blocking io\n");
     }
 
+    con = SSL_new(ctx);
     if (con == NULL) {
-        con = SSL_new(ctx);
+	ret = -1;
+	goto err;
```
Comment (dot-asm) [inline apps/s_server.c]: Formatting nit, replace tabs with spaces.

Comment (KaoruToda) [reply apps/s_server.c]: @dot-asm Thank you for your review. I replaced tabs with spaces.

File: apps/s_server.c [hunk 3]
```diff
@@ -2202,22 +2202,25 @@ static int sv_body(int s, int stype, int prot, unsigned char *context)
...
+    }
 
-        if (context
-            && !SSL_set_session_id_context(con,
-                                           context, strlen((char *)context))) {
-            BIO_printf(bio_err, "Error setting session id context\n");
-            ret = -1;
-            goto err;
-        }
+    if (context
```
Comment (dot-asm) [inline apps/s_server.c]: While you are at it, change to explicit `context != NULL`.

Comment (KaoruToda) [reply apps/s_server.c]: @dot-asm Thank you for review.
I change explicit "context != NULL".

**Comment by richsalz** [comment]
reconfirmed.

**Comment by dot-asm** [comment]
Merged. Thanks!

**Pull request closed without merging**